presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -767,6 +767,9 @@ private Object nullInputEvaluator(final Collection<Symbol> nullSymbols, Expressi
                     if (comparison.getType() == ComparisonExpression.Type.EQUAL) {
                         Set<Symbol> symbols1 = DependencyExtractor.extractUnique(comparison.getLeft());
                         Set<Symbol> symbols2 = DependencyExtractor.extractUnique(comparison.getRight());
+                        if (symbols1.isEmpty() || symbols2.isEmpty()) {
```

Comment:
Yes, it does. And unfortunately no.

Before this change, in case of comparison like `4 = a` (where `a` comes from left side of join), this method will return true, This would mean that method is a join equality comparison. Then in `visitJoin`, lines 379:
```
 boolean alignedComparison = Iterables.all(DependencyExtractor.extractUnique(equality.getLeft()), in(node.getLeft().getOutputSymbols()));
```
will say that this is an aligned comparison (left comparison symbols come from left side of join, right symbols from right) as left symbols comparison (empty set) are within left side symbols. While it is not true, as left side of comparison does not use any join symbols and right side of comparison is using symbol from left side of join.

I cannot add simple SQL test for that as usually this constant expression get projected. However, tests for rewrite for correlated subqueries will fail without this change.